exam/models/exam.py

```diff
@@ -81,6 +110,62 @@ def _check_date(self):
                                        invalid date!'))
         return True
 
+    @api.constrains('teacher_id')
+    def check_supervisior_exam(self):
+            for rec in self:
+                if rec.table_id.timetable_type == 'exam':
+                    if not rec.teacher_id:
+                        raise ValidationError(_('''Enter Supervisior'''))
+
+    @api.constrains('start_time', 'end_time')
+    def check_time(self):
+        for rec in self:
+            if rec.start_time >= rec.end_time:
+                raise ValidationError(_('''Start time should be
+                                        less than end time'''))
```

Comment:
Correct the error as : Configure the start time as earlier than end time.

Reply:
done